### ISSUE TITLE
Upgrade node-fetch due to vulnerability

### ISSUE THREAD

**Issue by grapevineai** [pouchdb/pouchdb#8281]
Is it possible to update node-fetch? A simple audit gets you that:

```
node-fetch  <=2.6.0 || 3.0.0-beta.1 - 3.0.0-beta.8
Denial of Service - https://npmjs.com/advisories/1556
fix available via `npm audit fix --force`
Will install pouchdb@7.0.0, which is a breaking change
node_modules/node-fetch
  pouchdb  >=7.1.0
  Depends on vulnerable versions of node-fetch
  node_modules/pouchdb
  pouchdb-fetch  >=7.1.0
  Depends on vulnerable versions of node-fetch
  node_modules/pouchdb-fetch
    pouchdb-abstract-mapreduce  >=7.1.0
    Depends on vulnerable versions of pouchdb-fetch
    node_modules/pouchdb-abstract-mapreduce
    pouchdb-find  >=7.1.0
    Depends on vulnerable versions of pouchdb-fetch
    node_modules/pouchdb-find
```

**Comment by beruic** [comment]
@grapevineai as far as I can see, this has already been addressed in the master branch, which means that it will probably be be fixed in the next release.

**Comment by beruic** [comment]
Check https://github.com/pouchdb/pouchdb/blob/master/package.json#L54

**Comment by NickCarducci** [comment]
Anyone having a Denial of Service still with the latest version?  Says it is patched `Patched in    │ >=2.6.1 <3.0.0-beta.1|| >= 3.0.0-beta.9` in `npm audit`, but I'm using version release `"pouchdb":"7.2.2"`.  Can't imagine the `"node-fetch":"3.x.x"` touching previous versions pouch uses, but I don't know what else to assume other than [recent activity by `node-fetch`](https://github.com/node-fetch/node-fetch/issues/1279) did something, but @jimmywarting says `node-fetch@2` will be maintained.  Seems 7 days ago, 2 days before the cited issue above, they released `"node-fetch":"2.6.2"` Could their warning be off and **`2.6.2` is now required**?  [Seems that way in their changes](https://github.com/node-fetch/node-fetch/commit/152214ca2f6e2a5a17d71e4638114625d3be30c6), just before they moved to v3.
